vyper/semantics/types/module.py

```diff
@@ -559,3 +559,19 @@ def immutable_section_bytes(self):
     @cached_property
     def interface(self):
         return InterfaceT.from_ModuleT(self)
+
+    @cached_property
+    def is_stateless(self):
+        """
+        Determine whether ModuleT is stateless by examining its top-level
+        declarations. A module has state if it contains storage variables,
+        transient variables, or immutables, or if it includes a "uses" or
+        "initializes" declaration.
+        """
+        if len(self.initializes_decls) > 0 or len(self.uses_decls) > 0:
+            return False
+        if any(not v.is_constant for v in self.variable_decls):
+            return False
+        if self.init_function is not None:
```

Comment:
does an init function determine the statefulnes of a module? i'd probably lean towards renaming the property